plugins/terraform/data_source_layer0_deploy.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/cli/client"
+)
+
+func dataSourceLayer0Deploy() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceLayer0DeployRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func datasourceLayer0DeployRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(client.Client)
+
+	deployName := d.Get("name").(string)
+	version := d.Get("version").(string)
+
+	deployID, err := resolveTags(client, deployName, map[string]string{
+		"type":    "deploy",
+		"version": version,
+	})
+	if err != nil {
+		return err
+	}
+
+	deploy, err := client.GetDeploy(deployID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(deploy.DeployID)
+
+	return setResourceData(d.Set, map[string]interface{}{
+		"id":      deploy.DeployID,
```

Comment:
Yeah, the blank value thing doesn't make sense. I think you got it correct on line 48, and should remove line 51. At the very least line 51 is redundant, but it could potentially cause issues treating it as a non-special field.